### PR TITLE
Cache the API server's parsed OpenAPI schema.

### PR DESCRIPTION
**Pull request by pgavlin**
Just what it says on the tin. This eliminates the last major source of
execution time in the k8s provider.

Contributes to #827.

### CODE REVIEW

**Comment by pgavlin** [comment]
@lblackstone is it safe to cache this data so early? Not sure how this interacts with CRDs, for example.

File: pkg/provider/provider.go [hunk 1]
```diff
@@ -100,13 +102,15 @@ type kubeProvider struct {
...
 	k8sVersion cluster.ServerVersion
+	resources  k8sopenapi.Resources
 }
 
 var _ pulumirpc.ResourceProviderServer = (*kubeProvider)(nil)
 
 func makeKubeProvider(
 	host *provider.HostClient, name, version string,
 ) (pulumirpc.ResourceProviderServer, error) {
+	cmdutil.InitProfiling("k8s-provider")
```
Comment (pgavlin) [inline pkg/provider/provider.go]: This and the other profiling-related change were accidental. I will remove them prior to merging.

File: pkg/openapi/openapi.go [hunk 2]
```diff
@@ -75,11 +70,11 @@ func ValidateAgainstSchema(
 	return specValidator.ValidateBytes(bytes)
 }
 
-// PatchForResourceUpdate introspects on the OpenAPI spec exposed by some client, and attempts to
-// generate a strategic merge patch for use in a resource update. If there is no specification of
-// how to generate a strategic merge patch, we fall back to JSON merge patch.
-func PatchForResourceUpdate(client discovery.CachedDiscoveryInterface, lastSubmitted, currentSubmitted,
-	liveOldObj *unstructured.Unstructured,
+// PatchForResourceUpdate introspects on the gevn OpenAPI spec and attempts to generate a strategic merge patch for
```
Comment (lblackstone) [inline pkg/openapi/openapi.go]:
```suggestion
// PatchForResourceUpdate introspects on the given OpenAPI spec and attempts to generate a strategic merge patch for
```

File: pkg/openapi/openapi.go [hunk 3]
```diff
@@ -97,11 +92,6 @@ func PatchForResourceUpdate(client discovery.CachedDiscoveryInterface, lastSubmi
 		return nil, "", nil, err
 	}
 
-	resources, err := getResourceSchemasForClient(client)
```
Comment (lblackstone) [inline pkg/openapi/openapi.go]: Will likely need to add a check that refreshes the cache on failure or following a CRD registration. If a CRD is registered after the schemas are cached, the cache won't include that info.

Comment (pgavlin) [reply pkg/openapi/openapi.go]: Done.

**Comment by lblackstone** [comment]
Just need a changelog, and this should be good to merge.

**Pull request merged**